### PR TITLE
[JENKINS-72505] f:validateButton finds selected radio button

### PR DESCRIPTION
**Pull request by timja**
See [JENKINS-72505](https://issues.jenkins.io/browse/JENKINS-72505).

Previously it just found the last button when doing validation that involved a radio button, it should try and find a selected one instead... 

Noticed in https://github.com/jenkinsci/azure-vm-agents-plugin/pull/493

### Testing done

Unit test, if you run it on master you get:
```
   4.505 [id=25]	WARNING	h.i.i.InstallUncaughtExceptionHandler#handleException: Caught unhandled exception with ID 03a79f27-87f9-4111-a5bc-403a8e405e76
org.junit.ComparisonFailure: expected:<[]f> but was:<[reallynot]f>
```

Manually:

Tested with a validateButton when editing a template in an azure cloud https://github.com/jenkinsci/azure-vm-agents-plugin/pull/493.
Previously the last radio button was being sent to the method handling the validate button, now it's the selected radio.
I also added an extra radio button locally to make sure it wasn't just taking the first one in the dom by continuing all the way.

### Proposed changelog entries

- JENKINS-72505, Find selected radio option when validating instead of the last one

### Proposed upgrade guidelines

N/A

```[tasklist]
### Submitter checklist
- [ ] The Jira issue, if it exists, is well-described.
- [ ] The changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developers, depending on the change) and are in the imperative mood (see [examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)). Fill in the **Proposed upgrade guidelines** section only if there are breaking changes or changes that may require extra steps from users during upgrade.
- [ ] There is automated testing or an explanation as to why this change has no tests.
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadocs, as appropriate.
- [ ] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")`, if applicable.
- [ ] New or substantially changed JavaScript is not defined inline and does not call `eval` to ease future introduction of Content Security Policy (CSP) directives (see [documentation](https://www.jenkins.io/doc/developer/security/csp/)).
- [ ] For dependency updates, there are links to external changelogs and, if possible, full differentials.
- [ ] For new APIs and extension points, there is a link to at least one consumer.
```

### Desired reviewers

@mention

Before the changes are marked as `ready-for-merge`:

```[tasklist]
### Maintainer checklist
- [ ] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [ ] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [ ] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [ ] Proper changelog labels are set so that the changelog can be generated automatically.
- [ ] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).
```

### CODE REVIEW

**Review by NotMyFault — APPROVED** [review]

File: test/src/test/java/lib/form/ValidateButtonTest.java [hunk 1]
```diff
@@ -109,13 +109,16 @@ public static final class DescriptorImpl extends Descriptor<TestValidateIsCalled
 
             public void doValidateTest1(@QueryParameter("a") String a, @QueryParameter("b") boolean b,
                                         @QueryParameter("c") boolean c, @QueryParameter("d") String d,
-                                        @QueryParameter("e") String e) {
+                                        @QueryParameter("e") String e,
+                                        @QueryParameter("f") String f
+                                        ) {
```
Comment (aneveux) [inline test/src/test/java/lib/form/ValidateButtonTest.java:114]:
```suggestion
                                        @QueryParameter("e") String e, @QueryParameter("f") String f) {
```

Nitpick for consistency. Feel free to ignore.

(isn't spotless enabled for core?)

**Review by aneveux — APPROVED** [review]
LGTM, thanks a lot for the fix!

**Review by MarkEWaite — APPROVED** [review]
This PR is now ready for merge.  We will merge it after approximately 24 hours if there is no negative feedback.

/label ready-for-merge

File: test/src/test/java/lib/form/ValidateButtonTest.java (hunk 1 above)
Comment (timja) [reply test/src/test/java/lib/form/ValidateButtonTest.java:114]: yes but it's less enabled than in plugins:
https://github.com/jenkinsci/jenkins/blob/b83c92666b5005d4c6e0b01cfda6c409a30bcf5d/pom.xml#L361-L373

vs

https://github.com/jenkinsci/plugin-pom/blob/6949f8af159d30e2722fcf773b397bc870e47359/pom.xml#L665

**Pull request merged**